py-polars/polars/functions/range.py

```diff
@@ -384,39 +384,44 @@ def date_range(
     name: str | None = None,
 ) -> Series | Expr:
     """
-    Create a range of type `Datetime` (or `Date`).
+    Generate a date range.
 
     Parameters
     ----------
     start
-        Lower bound of the date range, given as a date, datetime, Expr, or column name.
+        Lower bound of the date range.
     end
-        Upper bound of the date range, given as a date, datetime, Expr, or column name.
+        Upper bound of the date range.
     interval
-        Interval of the range periods; can be a python timedelta object like
-        ``timedelta(days=10)`` or a polars duration string, such as ``3d12h4m25s``
-        (representing 3 days, 12 hours, 4 minutes, and 25 seconds). Append
-        ``_saturating`` to the interval string to restrict resulting invalid dates to
-        valid ranges.
+        Interval of the range periods, specified as a Python ``timedelta`` object
+        or a Polars duration string like ``1h30m25s``.
+
+        Append ``_saturating`` to the interval string to restrict resulting invalid
+        dates to valid ranges.
 
         It is common to attempt to create a month-end date series by using the "1mo"
         offset string with a start date at the end of the month. This will not produce
         the desired results. See Note #2 below for further information.
     closed : {'both', 'left', 'right', 'none'}
-        Define whether the temporal window interval is closed or not.
+        Define which sides of the range are closed (inclusive).
     time_unit : {None, 'ns', 'us', 'ms'}
-        Set the time unit. Only takes effect if output is of ``Datetime`` type.
-    time_zone:
-        Optional timezone. Only takes effect if output is of ``Datetime`` type.
+        Time unit of the resulting ``Datetime`` data type.
+        Only takes effect if the output column is of type ``Datetime``.
+    time_zone
+        Time zone of the resulting ``Datetime`` data type.
+        Only takes effect if the output column is of type ``Datetime``.
     eager
-        Evaluate immediately and return a ``Series``. If set to ``False`` (default),
-        return an expression instead.
+        Evaluate immediately and return a ``Series``.
+        If set to ``False`` (default), return an expression instead.
     name
         Name of the output column.
 
         .. deprecated:: 0.18.0
             This argument is deprecated. Use the ``alias`` method instead.
 
+    Returns
+    -------
+    Column of data type ``Date`` or ``Datetime``.
```

Comment:
this was already there, but I've noticed it doesn't look great in the docs: https://pola-rs.github.io/polars/py-polars/html/reference/expressions/api/polars.date_range.html

Perhaps
```
Returns
-------
Series | Expr
    Column of data type ``Date`` or ``Datetime``.
```
is more what `Sphinx` is expecting?

Reply:
I'll check and make sure to update this so it gets rendered properly!

Reply:
Indeed, numpy docstyle requires the type in the returns section. That's unfortunate as the type hints already show those types. But yeah, we'll have to include it for Sphinx purposes. I do prefer human-readable format here, though (`or` instead of `|`):

```
    Returns
    -------
    Expr or Series
        Column of data type ``Int64``.
```

I also tried using :class:`Expr` but the result isn't any prettier/more readable and isn't clickable. So just `Expr or Series` is best, in my opinion.